### ISSUE TITLE
Renaming the units to their natural spelling

### ISSUE THREAD

**Issue by JohelEGP** [nholthaus/units#159]
For example, renaming `meter_t` to `meters`. This has probably been discussed in various places. I'll link to them as I find them.

To answer https://github.com/nholthaus/units/pull/150#issuecomment-405940556, I believe that, in the long run, renaming the units to their natural spelling is the right way to go. The `_t` suffix already has some uses that don't match the use in this library's units, like in the the standard type traits and the arguably more similar use in the types of `<cstdint>`. As the library becomes more popular, the `_t` suffix will become more of a baggage than the natural spelling the library can exploit with its CTAD capabilities and further possible improvements.

**Comment by nholthaus** [comment]
I agree with the logic. v3.0 is going to have a hefty upgrade burden on the users from 2.3, but that's the price of progress. We'll need an upgrade guide in the docs.

**Comment by johelegp** [comment]
What should we do about
https://github.com/nholthaus/units/blob/5483589322daad285eedd4e2822058ba4e6640d8/include/units/pressure.h#L75
and similar cases which have the same singular and plural name?

**Comment by johelegp** [comment]
[`dimensionless_unit`] also needs a name revision.

**Comment by nholthaus** [comment]
Done in 3.1.0: units use their natural plural spelling (`meters`); CTAD makes the angle brackets optional. Closing.
